### PR TITLE
Use 'FileLocator' instead of 'Platform'

### PR DESCRIPTION
**Pull request by sschulz92**

### CODE REVIEW

File: plugins/org.locationtech.udig.catalog.jgrass/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -15,7 +15,8 @@ Require-Bundle: org.eclipse.ui,
  org.locationtech.udig.catalog.ui,
  org.locationtech.udig.project.ui,
  org.locationtech.udig.mapgraphic,
- org.locationtech.udig.style
+ org.locationtech.udig.style,
+ org.junit
```
Comment (fgdrf) [inline plugins/org.locationtech.udig.catalog.jgrass/META-INF/MANIFEST.MF:19]: Not a good idea to depent on a test-library at runtime ;)

What is the reason to add it?

File: plugins/org.locationtech.udig.catalog.jgrass/src/org/locationtech/udig/catalog/jgrass/core/JGrassServiceTest.java [hunk 2]
```diff
@@ -9,97 +9,101 @@
...
+import org.eclipse.core.runtime.FileLocator;
+import org.junit.Test;
+import org.locationtech.udig.catalog.CatalogPlugin;
+import org.locationtech.udig.catalog.IGeoResourceInfo;
+import org.locationtech.udig.catalog.IResolve;
+import org.locationtech.udig.catalog.IService;
+import org.locationtech.udig.catalog.IServiceFactory;
+import org.locationtech.udig.catalog.jgrass.JGrassPlugin;
+
+public class JGrassServiceTest {
```
Comment (fgdrf) [inline plugins/org.locationtech.udig.catalog.jgrass/src/org/locationtech/udig/catalog/jgrass/core/JGrassServiceTest.java:30]: wondering why this class is in "production" bundle and not in a seperate test module... 

I guess this is the reason why the whole content was a comment :P

**Review by fgdrf — CHANGES_REQUESTED** [review]
Lets move the test-class into a catalog.jgrass.tests module to run it in an OSGi environment

File: plugins/org.locationtech.udig.catalog.jgrass/META-INF/MANIFEST.MF (hunk 1 above)
Comment (sschulz92) [reply plugins/org.locationtech.udig.catalog.jgrass/META-INF/MANIFEST.MF]: Was required to have JUnit available for the test case, but I moved the test case to the test bundle - therefore not required anymore :)

File: plugins/org.locationtech.udig.catalog.jgrass/src/org/locationtech/udig/catalog/jgrass/core/JGrassServiceTest.java (hunk 2 above)
Comment (sschulz92) [reply plugins/org.locationtech.udig.catalog.jgrass/src/org/locationtech/udig/catalog/jgrass/core/JGrassServiceTest.java]: I moved the JUnit test, you may have another look :)

**Comment by fgdrf** [comment]
Let me help here, I'd like to get it into a seperate module catalog.jgrass.tests which has a dependency to junit bundle and buindle catalog.jgrass

@sschulz92 I'll provide a fix for your branch, would be great if you can reviews it :)

File: plugins/org.locationtech.udig.catalog.jgrass/src/org/locationtech/udig/catalog/jgrass/core/JGrassServiceTest.java (hunk 2 above)
Comment (fgdrf) [reply plugins/org.locationtech.udig.catalog.jgrass/src/org/locationtech/udig/catalog/jgrass/core/JGrassServiceTest.java]: @sschulz92 Please have a look at changes in #531 which adresses the move of the test-case into a new bundle with test-dataset

File: plugins/org.locationtech.udig.catalog.jgrass.tests/test-data/jgrass-test/newmapset/hist/ArcGrid [hunk 3]
```diff
@@ -0,0 +1,8 @@
+Sat Oct 02 16:38:04 CEST 2021
+ArcGrid
+C:\dev\data\geodata\testdata\jgrass-db\jgrass-test\newmapset
+fgdrf
+DCELL
+
+
+Created by imageio-ext enabled JGrass
```
Comment (fgdrf) [inline plugins/org.locationtech.udig.catalog.jgrass.tests/test-data/jgrass-test/newmapset/hist/ArcGrid:8]: file content is empty on master ;) please merge again

File: plugins/org.locationtech.udig.catalog.tests/.classpath [hunk 4]
```diff
@@ -2,6 +2,10 @@
 <classpath>
 	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
 	<classpathentry kind="con" path="org.eclipse.pde.core.requiredPlugins"/>
-	<classpathentry kind="src" path="src/"/>
+	<classpathentry kind="src" path="src/">
+		<attributes>
+			<attribute name="test" value="true"/>
+		</attributes>
+	</classpathentry>
```
Comment (fgdrf) [inline plugins/org.locationtech.udig.catalog.tests/.classpath:9]: What option lead to the classpath entry, just "Plug-In Tools -> Update classpath"?

**Review by fgdrf — CHANGES_REQUESTED** [review]

File: plugins/org.locationtech.udig.catalog.tests/.classpath (hunk 4 above)
Comment (sschulz92) [reply plugins/org.locationtech.udig.catalog.tests/.classpath:9]: Yes, exactly :) Should I revert this change as it does not belong to the PR anymore.

Comment (fgdrf) [reply plugins/org.locationtech.udig.catalog.tests/.classpath:9]: No, please at it as it is and IMHO it belongs to this changeset.

File: plugins/org.locationtech.udig.catalog.jgrass.tests/test-data/jgrass-test/newmapset/hist/ArcGrid (hunk 3 above)
Comment (fgdrf) [reply plugins/org.locationtech.udig.catalog.jgrass.tests/test-data/jgrass-test/newmapset/hist/ArcGrid]: Hmm, looks like auto-merge failed, just remove the lines manually or replace file with latest from master. :)

**Review by fgdrf — APPROVED** [review]

**Pull request merged**